forge-gui/res/cardsfolder/upcoming/fishers_talent.txt

```diff
@@ -0,0 +1,15 @@
+Name:Fisher's Talent
+ManaCost:2 G U
+Types:Enchantment Class
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigPeek | TriggerDescription$ At the beginning of your upkeep, look at the top card of your library. You may reveal it if it's a land card. Create a 1/1 blue Fish creature token if you revealed it this way. Then draw a card.
+SVar:TrigPeek:DB$ PeekAndReveal | PeekAmount$ 1 | RevealValid$ Land | RevealOptional$ True | RememberRevealed$ True | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenAmount$ 1 | ConditionPresent$ Card.IsRemembered | TokenScript$ u_1_1_fish | TokenOwner$ You | SubAbility$ DBDraw

```

Comment:
your condition currently checks the battlefield

Reply:
Use `ConditionDefined$ Remembered`